app/components/search/SearchMap/SearchMap.tsx

```diff
@@ -17,34 +17,32 @@ import config from "../../../config";
 
 export const SearchMap = ({
   hits,
-  mapObject,
-  setMapObject,
-  setAroundLatLng,
   mobileMapIsCollapsed,
 }: {
   hits: TransformedSearchHit[];
-  mapObject: google.maps.Map | null;
-  setMapObject: (map: any) => void;
-  setAroundLatLng: (latLng: { lat: number; lng: number }) => void;
   mobileMapIsCollapsed: boolean;
 }) => {
-  const { userLocation } = useAppContext();
-  if (userLocation === null) {
-    return (
-      <div className="mapLoaderContainer">
-        <Loader />
-      </div>
-    );
-  }
+  const [googleMapObject, setMapObject] = useState<google.maps.Map | null>(
+    null
+  );
+  const { userLocation, aroundLatLng } = useAppContext();
+  const { setAroundLatLng } = useAppContextUpdater();
 
   function handleSearchThisAreaClick() {
-    const center = mapObject?.getCenter() || null;
+    const center = googleMapObject?.getCenter();
     if (center?.lat() && center?.lng()) {
-      setAroundLatLng({ lat: center.lat(), lng: center.lng() });
+      setAroundLatLng(`${center.lat()}, ${center.lng()}`);
     }
   }
 
-  const { lat, lng } = userLocation;
+  const { lat = 0, lng = 0 } = userLocation || {};
+  const aroundLatLngToMapCenter = {
+    lat: Number(aroundLatLng.split(",")[0]),
+    lng: Number(aroundLatLng.split(",")[1]),
+  };
+
+  // Center the map to the user's choice with a fallback to their location
+  const center = aroundLatLng ? aroundLatLngToMapCenter : { lat, lng };
```

Comment:
We need this logic now because there's a better distinction between the user location that we fetch on app load and the location the user sets in the map.